docker/autoware-universe/Dockerfile

```diff
@@ -42,6 +42,12 @@ RUN rm -rf \
   /etc/apt/sources.list.d/docker.list \
   /etc/apt/sources.list.d/nvidia-docker.list
 
+## Register Vulkan GPU vendors
```

Comment:
[Question] Will this problem only occur with Docker? If not, I think it's better to add this as an Ansible role to ensure the same environment for local and Docker development.

Reply:
It seems the NVIDIA repository was changed yesterday and https://gitlab.com/nvidia/container-images/vulkan/raw/master/nvidia_icd.json was removed.
https://gitlab.com/nvidia/container-images/vulkan/-/commit/19623df20b44a62315d262dedb7e6833daf10986

I think we have to change the approach. Please let me mark this as a draft.

Reply:
We can replace "master" by the commit before the change (https://gitlab.com/nvidia/container-images/vulkan/-/raw/dc389b0445c788901fda1d85be96fd1cb9410164/nvidia_icd.json). It was not a good idea to target the tip of the repository anyway.
Do you have other concerns for wanting to change the approach?

Reply:
> Do you have other concerns for wanting to change the approach?

Yes, I'd like to clarify the following items:
- In the first place, for what do we need `vulkan`?
- Before installing these settings, don't we need to install `vulkan` to the Docker images first? Currently, it's not installed yet.

Reply:
What kind of test are you thinking of?
From what I can see, the other steps of the Dockerfile are not tested except on "build-and-push" releases. Failure to download the new files would show up at the same time as failures in the the other steps, which sounds reasonable to me.
And we can't test that Nvidia GPUs are actually picked up by Vulkan in CI (unless there is a GPU attached on some runner?).

Reply:
I meant manual testing.
How can I confirm whether this PR works?

For example, like this:

```bash
cd ~/autoware
gh pr checkout 2736
./docker/build.sh
docker run --rm -it --gpus all ghcr.io/autowarefoundation/autoware-universe:humble-latest-cuda
apt update && apt install -y vulkan-tools
Run vulkaninfo or vkcube?
```

Reply:
> Failure to download the new files would show up at the same time as failures in the the other steps, which sounds reasonable to me.

By the way, what did you mean by this sentence? I'm sorry but I couldn't understand well.

Reply:
> I meant manual testing.
> How can I confirm whether this PR works?

Yes, `vulkan-tools` is a good way to test.
`vkcube` obviously requires X11 to work, but `vulkaninfo` also seems to require it when running from a local machine (but somehow not when testing it on a machine over ssh without X forwarding). Either way, I could test them both by launching (with `galactic` in my case):
```sh
docker run --rm -it --gpus all -e NVIDIA_DRIVER_CAPABILITIES=all -e DISPLAY -v /tmp/.X11-unix:/tmp/.X11-unix:rw -v $HOME/.Xauthority:$HOME/.Xauthority:rw -e XAUTHORITY=$HOME/.Xauthority --net=host ghcr.io/autowarefoundation/autoware-universe:galactic-latest-cuda
```
Although `vkcube` will work even without any hardware acceleration because `lavalpipe` is installed in the docker image. So you may want to check that it actually runs on the GPU with `nvidia-smi`.

> what did you mean by this sentence?

My point was that there was no need to individually test the lines I add in this patch.
If the download link becomes invalid, the Dockerfile `ADD ...` lines I add in this patch will make the `docker build` command fail, which will result in the `docker-build-and-push-*` CI job failing. And that would be good enough testing since it is good enough for the current Dockerfile steps.